04-vue-cli/03-UiDropdown1/components/UiDropdown.vue

```diff
@@ -1,31 +1,106 @@
 <template>
-  <div class="dropdown dropdown_opened">
-    <button type="button" class="dropdown__toggle dropdown__toggle_icon">
-      <UiIcon icon="tv" class="dropdown__icon" />
-      <span>Title</span>
+  <div class="dropdown" :class="{ dropdown_opened: isOpened }">
+    <button
+      type="button"
+      class="dropdown__toggle"
+      :class="{ dropdown__toggle_icon: hasIcon }"
+      @click="
+        () => {
+          isOpened = !isOpened;
+        }
+      "
```

Comment:
`v-on` directive wraps its content into a function. We don't need to provide a new one function and create it on each render.

`@click="isOpened = !isOpened"`